src/scripts/main.js

```diff
@@ -1,3 +1,230 @@
+      id: 'office',
+      type: 'select',
+      required: true,
+      dataQa: 'office',
+    },
+    {
+      label: 'Age',
+      id: 'age',
+      type: 'number',
+      required: true,
+      dataQa: 'age',
+    },
+    {
+      label: 'Salary',
+      id: 'salary',
+      type: 'number',
+      dataQa: 'salary',
+    },
+  ];
+
+  fields.forEach((field) => {
+    const label = document.createElement('label');
+
+    label.setAttribute('for', field.id);
+    label.textContent = `${field.label}:`;
+
+    let input;
+
+    if (field.type === 'select') {
+      input = document.createElement('select');
+      input.setAttribute('id', field.id);
+      input.setAttribute('name', field.id);
+      input.setAttribute('required', field.required);
+      input.setAttribute('data-qa', field.dataQa);
+
+      const options = [
+        'Tokyo',
+        'Singapore',
+        'London',
+        'New York',
+        'Edinburgh',
+        'San Francisco',
+      ];
+
+      options.forEach((optionText) => {
+        const option = document.createElement('option');
+
+        option.value = optionText;
+        option.textContent = optionText;
+        input.appendChild(option);
+      });
+    } else {
+      input = document.createElement('input');
+      input.setAttribute('type', field.type);
+      input.setAttribute('id', field.id);
+      input.setAttribute('name', field.id);
+      input.setAttribute('data-qa', field.dataQa);
+    }
+
+    label.appendChild(input);
+    form.appendChild(label);
+  });
+
+  const submitButton = document.createElement('button');
+
+  submitButton.type = 'submit';
+  submitButton.textContent = 'Save to table';
+  form.appendChild(submitButton);
+
+  formContainer.appendChild(form);
+  tableContainer.appendChild(formContainer);
+
+  form.addEventListener('submit', (e) => {
+    e.preventDefault();
+
+    const nam = document.getElementById('name').value.trim();
+    const position = document.getElementById('position').value.trim();
+    const office = document.getElementById('office').value;
+    const age = parseInt(document.getElementById('age').value);
+    const salary = parseInt(document.getElementById('salary').value);
+
+    let isValid = true;
+
+    clearNotifications();
+
+    if (nam.length < 4) {
+      showNotification('error', 'Name must have at least 4 characters.');
+      isValid = false;
+    }
+
+    if (!/^[a-zA-Zа-яА-ЯёЁ]+$/.test(nam)) {
+      showNotification('error', 'Name must contain only letters.');
+      isValid = false;
+    }
+
+    if (position === '' || position.length < 4) {
+      showNotification('error', 'Position must have at least 4 characters.');
+      isValid = false;
+    }
+
+    if (isNaN(age)) {
+      showNotification('Age is not a valid number');
+      isValid = false;
+    } else if (age < 18 || age > 90) {
+      showNotification('Age must be between 18 and 90');
+      isValid = false;
+    } else {
+      showNotification('Age is valid:', age);
```

Comment:
The notification message for valid age is inconsistent with other notifications. Consider removing this notification or changing it to an 'info' type if you want to keep it.